h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -696,10 +693,63 @@ public void init(boolean expensive) {
         _parms._gradient_epsilon = _parms._lambda[0] == 0 ? 1e-6 : 1e-4;
         if(_parms._lambda_search) _parms._gradient_epsilon *= 1e-2;
       }
+
+      if (_parms.hasCheckpoint()) {
+        if (!Family.gaussian.equals(_parms._family))  // Gaussian it not iterative and therefore don't care
+          _checkPointFirstIter = true;  // mark the first iteration during iteration process of training
+        if (!Solver.IRLSM.equals(_parms._solver))
+          error("_checkpoint", "GLM checkpoint is supported only for IRLSM.  Please specify it " +
+                  "explicitly.  Do not use AUTO or default");
+        Value cv = DKV.get(_parms._checkpoint);
+        CheckpointUtils.getAndValidateCheckpointModel(this, _parms.CHECKPOINT_NON_MODIFIABLE_FIELDS, cv);
```

Comment:
referecing a static field from a non-static context, you can do just a static import here, much nicer